src/frequenz/sdk/config/_manager.py

```diff
@@ -133,26 +164,72 @@ def new_receiver(
             ```
 
         Args:
+            key: The configuration key to be read by the receiver.
             skip_unchanged: Whether to skip sending the configuration if it hasn't
                 changed compared to the last one received.
 
         Returns:
             The receiver for the configuration.
         """
-        receiver = self.config_channel.new_receiver(name=str(self), limit=1)
+        receiver = self.config_channel.new_receiver(name=f"{self}:{key}", limit=1)
+
+        def _get_key_or_error(
+            config: Mapping[str, Any]
+        ) -> Mapping[str, Any] | InvalidValueForKeyError | None:
+            try:
+                return _get_key(config, key)
+            except InvalidValueForKeyError as error:
+                return error
+
+        key_receiver = receiver.map(_get_key_or_error)
```

Comment:
I think it would be more clear (and it would simplify other methods) if you finish processing `key` here completely ( so it doesn't influence other methods )
So could you please add extra filter: " IsNotError()"

```
def isNotErro(config):
   if isinstance(new_value, InvalidValueForKeyError):
      _logger.error
      return False
   return True 
```

 `key_receiver = receiver.map(_get_key_or_error).filter(isNotError)`

Reply:
This was changed in subsequent commits, does this comment still applies? Maybe you can make it in the end full diff instead so the context makes more sense?

Reply:
I meant something like this:
```
        receiver = self.config_channel.new_receiver(name=f"{self}:{key}", limit=1)
        receiver = receiver.map(_get_key_or_error).filter(is_not_exception)
        receiver = receiver.map(_load_config_or_error).filter(is_not_exception)
        if skip_unchanged:
            receiver = receiver.filter(WithPrevious[DataclassT | None](
                lambda old, new: _not_equal_with_logging(
                    key=key, old_value=old, new_value=new
                )
            ))
```
Then all following method will simplify , you don't have to pass Exception as argument and check for exception. And typing will simplify, too because filter(is_not_exception) will break as soon any exception appears.

But I just found outthat the receiver returns InvalidValueForKeyError exception, why is it necessary?
`Receiver[Mapping[str, Any] | InvalidValueForKeyError | None]:`

Reply:
The original PR did what you are suggesting (completely hide config loading errors from the downstream users), but I thought doing that might not be the best approach, maybe downstream actors would want to do something with this information (an error and what kind of error while trying to get the new configuration), like falling back to some safe default config or something, the current approach of keeping the old configuration is only one possible way to handle errors, but errors are still there and people might want to handle them differently.

If actors want to keep the old config if there are errors retrieving the new config, they can do it explicitly, it is just 2 extra lines of code. I also think it is good to see at the actor level more explicitly that we are handling possible errors.

I would at some point even make the `ConfigManagingActor` also send exceptions via the channel, like errors reading the file or parsing the TOML, that right now are completely being hidden to the users.

This was the main motivation for this.

Reply:
Ok, you are right, it is better to keep it.  Thanks for explanation!